codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/customize/CoreCodegenDecorator.kt

```diff
@@ -62,6 +65,31 @@ interface CoreCodegenDecorator<CodegenContext> {
         baseCustomizations: List<LibRsCustomization>,
     ): List<LibRsCustomization> = baseCustomizations
 
+    /**
+     * Hook to customize structures generated by `StructureGenerator`.
+     */
+    fun structureCustomizations(
+        codegenContext: CodegenContext,
+        baseCustomizations: List<StructureCustomization>,
+    ): List<StructureCustomization> = baseCustomizations
+
+    // TODO(https://github.com/awslabs/smithy-rs/issues/1401): Move these customizations into `ClientCodegenDecorator`
```

Comment:
cc @jjant for #2159.

Reply:
Also, only the first customization needs to be moved right? `errorImplCustomizations` is fed to `ErrorImplGenerator` which the server uses.

Reply:
Rephrased the TODO comment to limit it to just the builder customizations.